File: URI/Java/Media3.java
import java.util.Scanner;
import java.lang.Math;

public class Media3 {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        float n1, n2, n3, n4, media;
        n1 = input.nextFloat();
        n2 = input.nextFloat();
        n3 = input.nextFloat();
        n4 = input.nextFloat();

        media = (2*n1 + 3*n2 + 4*n3 + n4)/10.0f;
        String str = String.format("%.1f", media);
        System.out.println("Media: " + str);
        
        if (media >= 7.0f) {
            System.out.println("Aluno aprovado.");
        } else if (media < 5.0f) {
            System.out.println("Aluno reprovado.");
        } else if (media >= 5.0f && media <= 6.9f) {
            System.out.println("Aluno em exame.");
            float exame_final = input.nextFloat();
            media += exame_final;
            media /= 2.0f;
            String str2 = String.format("%.1f", exame_final);
            System.out.println("Nota do exame: "+str2);
            if (media >= 5.0f) {
                System.out.println("Aluno aprovado.");
            } else {
                System.out.println("Aluno reprovado.");
            }
            String str3 = String.format("%.1f", media);
            System.out.println("Media final: " + str3);
        }
        input.close();
    }
}

File: URI/Python/Media3.py
n = [float(x) for x in input().split()]
media = n[0]*2 + n[1]*3 + n[2]*4 + n[3]
media/= 10.0
print("Media: {:.1f}".format(media))
if media >= 7.0:
    print("Aluno aprovado.")
elif media < 5.0:
    print("Aluno reprovado.")
else:
    n1 = float(input())
    print("Aluno em exame.")
    print("Nota do exame: {:.1f}".format(n1))
    media = ( media + n1 )/2
    if media >= 5.0:
        print("Aluno aprovado.")
    else:
        print("Aluno reprovado.")
    print("Media final: {:.1f}".format(media))